src/main/java/com/disneystreaming/pg2k4j/SlotReaderKinesisWriter.java

```diff
@@ -253,18 +255,32 @@ public void resetIdleCounter() {
     }
 
     Stream<UserRecord> getUserRecords(final SlotMessage slotMessage) throws
-            JsonProcessingException {
-        Stream<ByteBuffer> byteBuffers = Stream.of(ByteBuffer.wrap(
-                objectMapper.writeValueAsBytes(slotMessage)));
-        return byteBuffers.map(
-                byteBuffer -> {
-                    Random r = new Random();
-                    return new UserRecord(streamName, Long.toString(System
-                            .currentTimeMillis()),
-                            new BigInteger(randomBigIntBits, r)
+            IOException {
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        try (DeflaterOutputStream dos = new DeflaterOutputStream(baos)) {
+            objectMapper.writeValue(dos, slotMessage);
+
+            byte[] bytes = baos.toByteArray();
+            if (bytes.length > KIBI * KIBI) {
```

Comment:
Tähän voisi satunnaista koodinlukijaa varten kommentoida, että raja on 1MB ja mistä päin internetsiä se käy ilmi